tfjs-backend-webgpu/perf/tune.js

```diff
@@ -0,0 +1,381 @@
+    // appends the Text Node we created into the cell <td>
+    mycurrent_cell.appendChild(currenttext);
+    // appends the cell <td> into the row <tr>
+    mycurrent_row.appendChild(mycurrent_cell);
+  });
+  // appends the row <tr> into <tbody>
+  mytablebody.appendChild(mycurrent_row);
+
+  // appends <tbody> into <table>
+  mytable.appendChild(mytablebody);
+  // appends <table> into <body>
+  mybody.appendChild(mytable);
+  // sets the border attribute of mytable to 2;
+  mytable.setAttribute("border", "2");
+
+  // Add sorting for tr
+  document.querySelectorAll('th').forEach(th => th.addEventListener('click', (() => {
+    const table = th.closest('table');
+    Array.from(table.querySelectorAll('tr:nth-child(n+2)'))
+      .sort(comparer(Array.from(th.parentNode.children).indexOf(th), this.asc = !this.asc))
+      .forEach(tr => table.appendChild(tr));
+  })));
+
+  run();
+}
+
+async function timeMatmul(rowNum) {
+  await tf.setBackend('webgpu');
+  const dimAOuter = parseInt(STATE.input.split(',')[0]);
+  const dimInner = parseInt(STATE.input.split(',')[1]);
+  const dimBOuter = parseInt(STATE.input.split(',')[2]);
+  const numIterations = STATE.iteration;
+  const tensorA = tf.tensor2d(
+    Array.from({ length: dimAOuter * dimInner }, () => Math.floor(Math.random())),
+    [dimAOuter, dimInner]
+  );
+  const tensorB = tf.tensor2d(
+    Array.from({ length: dimInner * dimBOuter }, () => Math.floor(Math.random())),
+    [dimInner, dimBOuter]
+  );
+  document.getElementById('message').innerHTML =
+    `Testing on webgpu, ${STATE.name}, ${STATE.input} ...`;
+
+  // skip the first time
+  await tf.matMul(tensorA, tensorB).data();
+
+  let profile_webgpu, profile_webgl;
+
+  for (let i = 0; i < numIterations; i++) {
+    profile_webgpu = await tf.profile(async () => {
+      await tf.matMul(tensorA, tensorB).data();
+    });
+  }
+  const kernel_webgpu = profile_webgpu.kernels[0];
+
+  await tf.setBackend('webgl');
+  document.getElementById('message').innerHTML =
+    `Testing on webgl, ${STATE.name}, ${STATE.input} ...`;
+  for (let i = 0; i < numIterations; i++) {
+    profile_webgl = await tf.profile(async () => {
+      await tf.matMul(tensorA, tensorB).data();
+    });
+  }
+  const kernel_webgl = profile_webgl.kernels[0];
+
+  tensorA.dispose();
+  tensorB.dispose();
+
+  const result = {};
+  result.uid = `${STATE.name}_${STATE.input}_${STATE.iteration}`
+  result.name = STATE.name;
+  result.backend = STATE.backend;
+  result.input = STATE.input;
+  result.iteration = STATE.iteration;
+  result.webgpu = `${parseFloat(kernel_webgpu.kernelTimeMs).toFixed(2)}`; // WebGPU
+  result.webgl = `${parseFloat(kernel_webgl.kernelTimeMs).toFixed(2)}`; // WebGL
+  result.webglComp = `${parseFloat((result.webgl / result.webgpu) * 100).toFixed(0)}`; // WebGLComp
+  result.scale = `${dimAOuter * dimInner * dimBOuter}`;  // scale
+  result.program = `${kernel_webgpu.extraInfo.split(':')[0]}`;  // scale
```

Comment:
nit: `// scale` wrong annotation for `result.program`?

Reply:
Thanks, done.